src/gui/editorwidgets/qgsfilenamewidgetwrapper.cpp

```diff
@@ -101,42 +136,132 @@ void QgsFileNameWidgetWrapper::initWidget( QWidget* editor )
 
 void QgsFileNameWidgetWrapper::setValue( const QVariant& value )
 {
+  mValue = value;
   if ( mLineEdit )
   {
     if ( value.isNull() )
+    {
       mLineEdit->setText( QSettings().value( "qgis/nullValue", "NULL" ).toString() );
+    }
     else
+    {
       mLineEdit->setText( value.toString() );
+    }
   }
 
   if ( mLabel )
-    mLabel->setText( value.toString() );
+  {
+    mLabel->setText( toUrl( value ) );
+    valueChanged( value.toString() ); // emit signal that value has changed
+  }
+
 }
 
-void QgsFileNameWidgetWrapper::selectFileName()
+void QgsFileNameWidgetWrapper::setEnabled( bool enabled )
 {
-  QString text;
-
   if ( mLineEdit )
-    text = mLineEdit->text();
+    mLineEdit->setReadOnly( !enabled );
 
-  if ( mLabel )
-    text = mLabel->text();
+  if ( mPushButton )
+    mPushButton->setEnabled( enabled );
+}
 
-  QString fileName = QFileDialog::getOpenFileName( mLineEdit, tr( "Select a file" ), QFileInfo( text ).absolutePath() );
+// Open a file/dir selector and affect the value to the widget
+void QgsFileNameWidgetWrapper::selectFileName()
+{
+  QSettings settings;
+  QString oldPath;
+
+  // If we use fixed default path
+  if ( config().contains( "DefaultRoot" ) )
+    oldPath = QDir::cleanPath( config( "DefaultRoot").toString() );
+  // if we use a relative path option, we need to obtain the full path
+  else if ( !value().isNull() )
+    oldPath = relativePath( value().toString(), false );
+  else
+  {
+    // If there is no valid value, find a default path to use
+    QUrl theUrl = QUrl::fromUserInput( oldPath );
```

Comment:
No, it is a mistake from code manipulations. oldPath is assigned NULL (well, [I think](https://stackoverflow.com/questions/9724966/how-to-initialize-a-qstring-to-null)).

The final else statement should be deleted and its content put outside of the initial test: we always need to evaluate if oldpath is valid or not (and not evaluate it only when the value is null). Would you like to fix this (or should I do it) ?